job_script_rnn.py --config bcrl.txt
========================================
Collecting expert demonstrations
========================================
========================================
Running 2-batch size RNN BC with expert demonstrations
========================================
========================================
BC training complete !!!
time taken = 22.460187
========================================
/home/flamingwizard4/anaconda3/envs/mjrl-env/lib/python3.7/site-packages/torch/nn/modules/loss.py:431: UserWarning: Using a target size (torch.Size([5, 30])) that is different to the input size (torch.Size([1, 5, 30])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
torch.Size([5, 1])
Traceback (most recent call last):
  File "job_script_rnn.py", line 146, in <module>
    train_losses, test_losses = e.evaluate_train_vs_test(bc_agent, demo_paths)
  File "/home/flamingwizard4/Desktop/mujoco_torch/mjrl/mjrl/utils/gym_env_rnn.py", line 244, in evaluate_train_vs_test
    losses = [agent.old_mse_loss(data_dict, idx=train_rdx), self_loss, p]
  File "/home/flamingwizard4/Desktop/mujoco_torch/mjrl/mjrl/algos/rnn_behavior_cloning.py", line 135, in old_mse_loss
    act_pi = torch.unsqueeze(act_pi, 1)
UnboundLocalError: local variable 'act_pi' referenced before assignment
(mjrl-env) flamingwizard4@ldd:~/Desktop/mujoco_torch/hand_dapg/dapg/examples$ python job_script_rnn.py --config bcrl.txt
========================================
Collecting expert demonstrations
========================================
========================================
Running 2-batch size RNN BC with expert demonstrations
========================================
========================================
BC training complete !!!
time taken = 23.740027
========================================
/home/flamingwizard4/anaconda3/envs/mjrl-env/lib/python3.7/site-packages/torch/nn/modules/loss.py:431: UserWarning: Using a target size (torch.Size([5, 30])) that is different to the input size (torch.Size([1, 5, 30])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
torch.Size([5, 1])
Traceback (most recent call last):
  File "job_script_rnn.py", line 146, in <module>
    train_losses, test_losses = e.evaluate_train_vs_test(bc_agent, demo_paths)
  File "/home/flamingwizard4/Desktop/mujoco_torch/mjrl/mjrl/utils/gym_env_rnn.py", line 244, in evaluate_train_vs_test
    losses = [agent.old_mse_loss(data_dict, idx=train_rdx), self_loss, p]
  File "/home/flamingwizard4/Desktop/mujoco_torch/mjrl/mjrl/algos/rnn_behavior_cloning.py", line 135, in old_mse_loss
    act_pi = torch.unsqueeze(act_pi, 1)
UnboundLocalError: local variable 'act_pi' referenced before assignment
